pkg/controller/job/job_controller_test.go

```diff
@@ -691,7 +716,7 @@ func TestControllerSyncJob(t *testing.T) {
 				if err == nil {
```

Comment:
I'm starting to dislike all this logic to determine if there should have been an error or not.

We should rather add a field to the test table `expectError bool` and perhaps put a comment on why we expect an error or not for each test case. Could be for a follow up PR though.

Reply:
Agree. maybe also add error message to the test table when there should be an error?

Reply:
Ideally, we should match the error that we expect, but perhaps a string comparisons is fine to start

Reply:
Then I will start another PR to modify the error match logic